### PR TITLE
chore: node 20 + pnpm 8

### LINKED ISSUE #4509: Upgrade pnpm so it won't error in nodejs 20
### System Info

-

### Details

pnpm/pnpm#6424

### Reproduce link

_No response_

### Reproduce Steps

-

### PR DESCRIPTION
**Pull request by Boshen**
closes #4509

### CODE REVIEW

**Comment by Boshen** [comment]
🙏 CI please pass

File: package.json [hunk 1]
```diff
@@ -68,5 +68,5 @@
     "sass-embedded-win32-ia32": "1.58.3",
     "sass-embedded-win32-x64": "1.58.3"
   },
-  "packageManager": "pnpm@7.32.0"
-}
\ No newline at end of file
+  "packageManager": "pnpm@v8.10.5"
```
Comment (hardfist) [inline package.json:71]: you need change engine.pnpm field too

Comment (Boshen) [reply package.json:71]: We don't have a engine field 🤔

Comment (hardfist) [reply package.json:71]: it seems we need to rebase 0.4.0 to main first

Comment (hardfist) [reply package.json:71]: now you have engine field

**Comment by Boshen** [comment]
@ahabhgk 

The test 

https://github.com/web-infra-dev/rspack/blob/e1934125d5e003e408f07403bb6ce89e1d379f24/packages/rspack-cli/tests/build/profile/profile.test.ts#L34-L41

is throwing 

```
 FAIL  tests/build/profile/profile.test.ts
  ● Console

    console.log
      Error: Command was killed with SIGABRT (Aborted): /Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack
      thread '<unnamed>' panicked at /Users/bytedance/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-subscriber-0.3.17/src/util.rs:91:14:
      failed to set global default subscriber: SetGlobalDefaultError("a global default trace dispatcher has already been set")
      note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
      fatal runtime error: failed to initiate panic, error 5
          at makeError (/Users/bytedance/github/rspack/node_modules/execa/lib/error.js:60:11)
          at handlePromise (/Users/bytedance/github/rspack/node_modules/execa/index.js:118:26)
          at processTicksAndRejections (node:internal/process/task_queues:95:5)
          at Object.<anonymous> (/Users/bytedance/github/rspack/packages/rspack-cli/tests/build/profile/profile.test.ts:35:13) {
        shortMessage: 'Command was killed with SIGABRT (Aborted): /Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack',
        command: '/Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack',
        escapedCommand: '"/Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack"',
        exitCode: undefined,
        signal: 'SIGABRT',
        signalDescription: 'Aborted',
        stdout: '',
        stderr: "thread '<unnamed>' panicked at /Users/bytedance/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-subscriber-0.3.17/src/util.rs:91:14:\n" +
          'failed to set global default subscriber: SetGlobalDefaultError("a global default trace dispatcher has already been set")\n' +
          'note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n' +
          'fatal runtime error: failed to initiate panic, error 5',
        failed: true,
        timedOut: false,
        isCanceled: false,
        killed: false
      }
```

What is going on 😅 ?

**Comment by hardfist** [comment]
> @ahabhgk
> 
> The test
> 
> https://github.com/web-infra-dev/rspack/blob/e1934125d5e003e408f07403bb6ce89e1d379f24/packages/rspack-cli/tests/build/profile/profile.test.ts#L34-L41
> 
> is throwing
> 
> ```
>  FAIL  tests/build/profile/profile.test.ts
>   ● Console
> 
>     console.log
>       Error: Command was killed with SIGABRT (Aborted): /Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack
>       thread '<unnamed>' panicked at /Users/bytedance/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-subscriber-0.3.17/src/util.rs:91:14:
>       failed to set global default subscriber: SetGlobalDefaultError("a global default trace dispatcher has already been set")
>       note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
>       fatal runtime error: failed to initiate panic, error 5
>           at makeError (/Users/bytedance/github/rspack/node_modules/execa/lib/error.js:60:11)
>           at handlePromise (/Users/bytedance/github/rspack/node_modules/execa/index.js:118:26)
>           at processTicksAndRejections (node:internal/process/task_queues:95:5)
>           at Object.<anonymous> (/Users/bytedance/github/rspack/packages/rspack-cli/tests/build/profile/profile.test.ts:35:13) {
>         shortMessage: 'Command was killed with SIGABRT (Aborted): /Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack',
>         command: '/Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack',
>         escapedCommand: '"/Users/bytedance/github/rspack/packages/rspack-cli/bin/rspack"',
>         exitCode: undefined,
>         signal: 'SIGABRT',
>         signalDescription: 'Aborted',
>         stdout: '',
>         stderr: "thread '<unnamed>' panicked at /Users/bytedance/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-subscriber-0.3.17/src/util.rs:91:14:\n" +
>           'failed to set global default subscriber: SetGlobalDefaultError("a global default trace dispatcher has already been set")\n' +
>           'note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n' +
>           'fatal runtime error: failed to initiate panic, error 5',
>         failed: true,
>         timedOut: false,
>         isCanceled: false,
>         killed: false
>       }
> ```
> 
> What is going on 😅 ?
> 
> I'm not sure whether we should:
> 
> * pin signal-exit?
> * fix the test?
> * fix the exit code?

seems same as https://github.com/web-infra-dev/rspack/pull/4245/files#r1395400223

**Pull request merged**